rust/otel-arrow-rust/src/encode/record/logs.rs

```diff
@@ -0,0 +1,751 @@
+            fields.push(Field::new(
+                consts::SCHEMA_URL,
+                array.data_type().clone(),
+                true,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.time_unix_nano.finish() {
+            fields.push(Field::new(
+                consts::TIME_UNIX_NANO,
+                array.data_type().clone(),
+                false,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.observed_time_unix_nano.finish() {
+            fields.push(Field::new(
+                consts::OBSERVED_TIME_UNIX_NANO,
+                array.data_type().clone(),
+                false,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.trace_id.finish() {
+            fields.push(Field::new(
+                consts::TRACE_ID,
+                array.data_type().clone(),
+                true,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.span_id.finish() {
+            fields.push(Field::new(consts::SPAN_ID, array.data_type().clone(), true));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.severity_number.finish() {
+            fields.push(Field::new(
+                consts::SEVERITY_NUMBER,
+                array.data_type().clone(),
+                true,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.severity_text.finish() {
+            fields.push(Field::new(
+                consts::SEVERITY_TEXT,
+                array.data_type().clone(),
+                true,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(body) = self.body.finish().transpose()? {
+            fields.push(Field::new(consts::BODY, body.data_type().clone(), true));
+            columns.push(Arc::new(body));
+        }
+
+        if let Some(array) = self.dropped_attributes_count.finish() {
+            fields.push(Field::new(
+                consts::DROPPED_ATTRIBUTES_COUNT,
+                array.data_type().clone(),
+                false,
+            ));
+            columns.push(array);
+        }
+
+        if let Some(array) = self.flags.finish() {
+            fields.push(Field::new(consts::FLAGS, array.data_type().clone(), true));
+            columns.push(array);
+        }
+
+        RecordBatch::try_new(Arc::new(Schema::new(fields)), columns)
+    }
+}
+
+impl Default for LogsRecordBatchBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Builder for the body of a log record.
+pub struct LogsBodyBuilder {
+    value_type: UInt8ArrayBuilder,
+    string_value: StringArrayBuilder,
+    int_value: Int64ArrayBuilder,
+    double_value: Float64ArrayBuilder,
+    bool_value: AdaptiveBooleanArrayBuilder,
+    bytes_value: BinaryArrayBuilder,
+    ser_value: BinaryArrayBuilder,
+    nulls: NullBufferBuilder,
```

Comment:
I'm (again) confused about why/how LogsBody is different from AnyValue. I see that nulls is different here and there. I wonder if the AnyValue should support nulls to make these cases the same?